### ISSUE TITLE
Non-POSIX (and generally broken) Syntax

### ISSUE THREAD

**Issue by terminalforlife** [huijunchen9260/dmenufm#24]
Line 146 in `dmenufm`:

```sh
if < "$(find "$XDGDIR1" "$XDGDIR2" -name "$appdesktop" \
```

I mentioned this in my last PR, but it seems it wasn't addressed. This is just outright broken. I wouldn't claim POSIX compliance until this and any other non-POSIX usages are addressed, honestly. A better wording could be "POSIX-mindful" or "POSIX-aware".

Even Bash in POSIX mode doesn't understand it. The `if` keyword doesn't, AFAIK, read from STDIN, which is why this is inherently broken.

**Comment by huijunchen9260** [comment]
The `<` part is not referred to the if, but to the grep. I can adjust the code and you can see:

```
if grep "Terminal=false" < "$(find "$XDGDIR1" "$XDGDIR2" -name "$appdesktop" | tail -n 1)" ; then
```

**Comment by terminalforlife** [comment]
![Selectshot_2019-12-17_22:22:27](https://user-images.githubusercontent.com/31768530/71039308-bb64cb00-211b-11ea-8e0d-c498d345286c.jpg)

**Comment by terminalforlife** [comment]
I think it's best to just write it the regular way, rather than confusing people looking at your code. When I look at that, I just see that you're wanting `if` to read from STDIN.

By the way, this is why I say it's non-POSIX:

![Selectshot_2019-12-17_22:27:38](https://user-images.githubusercontent.com/31768530/71039645-8d33bb00-211c-11ea-98a2-1eb9d2d62362.jpg)

That's yash in posixly-correct mode (good test for POSIX conformance, apparently). It _should_, if I'm not being blonde, be an exit status of 1, because that line is _not_ in my `.bashrc` file.

![Selectshot_2019-12-17_22:30:37](https://user-images.githubusercontent.com/31768530/71039774-e0a60900-211c-11ea-954d-12b291141c2b.jpg)

Now if I do the same thing in Bash in posixly-correct mode, the same thing happens.

**Comment by huijunchen9260** [comment]
If statement cannot be wrong. You are either execute the command and echo yes, or not successfully execute the command, and not echoing yes. 

Based your screenshot, the `< $(find...) grep "..."` part is returning 1, so your terminal is not printing "yes". `echo $?` is evaluating the whole if statement, so no matter `< $(find ...) grep "..."` part is correct or wrong, the if statement is always successfully echoing or not echoing yes.

Addressing to you "Actually," part, this is why no matter what command you put in the if statement, you always get `0` in `echo $?`.

I also downloaded `yash` and try to run my code with it, and everything seems to be fine.

Hope this comment can help you.

**Comment by terminalforlife** [comment]
Quite right, the if statement effectively protects the status from spilling out, which is why I used echo to test it, and in neither cases did it output `yes`. :/

As for `yash`, I guess I'll eat a good portion of my words. I've just done the same. I got a few errors, but I think it's just because of the assumptions made in dmenufm, rather than POSIX-incorrectedness. lol Sorry for the confusion.
